### PR TITLE
boards: stm32: Fix board vendor when required

### PR DESCRIPTION
**Pull request by erwango**
When vendor is unknown, no vendor is provided and compatible used is "board_name" instead of "vendor,board_name".

### CODE REVIEW

**Review by nashif — DISMISSED** [review]

File: boards/arm/black_f407ve/black_f407ve.dts [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 / {
 	model = "black_f407ve board";
-	compatible = "st,stm32f407";
+	compatible = "unknown,black-stm32f407";
```
Comment (henrikbrixandersen) [inline boards/arm/black_f407ve/black_f407ve.dts:13]: I think you should just leave out the vendor prefix, then - don't introduce an `unknown` prefix.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: boards/arm/black_f407ve/black_f407ve.dts (hunk 1 above)
Comment (erwango) [reply boards/arm/black_f407ve/black_f407ve.dts:13]: ok, but then, how should it be taken into account in https://github.com/zephyrproject-rtos/zephyr/pull/59642 ?

Comment (henrikbrixandersen) [reply boards/arm/black_f407ve/black_f407ve.dts:13]: I don't know, but adding an `unknown` vendor prefix is in violation with our binding rules: https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#rules-for-vendor-prefixes

Comment (erwango) [reply boards/arm/black_f407ve/black_f407ve.dts:13]: > If your device is not manufactured by a specific hardware vendor, do not invent a vendor prefix. Vendor prefixes are not mandatory parts of compatible properties, and compatibles should not include them unless they refer to an actual vendor. There are some exceptions to this rule, but the practice is strongly discouraged.

ok, clear. I was not aware, I'll update.
Then, @nashif should take it into account in #56642, as I guess a vendor is always expected.

Comment (erwango) [reply boards/arm/black_f407ve/black_f407ve.dts]: @henrikbrixandersen Fixed.
@nashif Please make sure you're considering the case where no vendor name is present in #59642

Comment (henrikbrixandersen) [reply boards/arm/black_f407ve/black_f407ve.dts]: Looks good, thanks. Please update the PR description as well.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**